buildSrc/docker.gradle

```diff
@@ -147,18 +148,100 @@ dockerImagesToBuild.each { imageName ->
     }
     project.tasks.findByName(dockerBuildAllImages).dependsOn dockerImageBuildTaskName
 
-    project.tasks.create(name: dockerImagePublishTaskName, type: Exec, group: 'Docker', description: "Publish ${imageName} docker image to hub.docker.com") {
-        dockerTasksCreated.add(dockerImagePublishTaskName)
-        dependsOn project.tasks.findByName(dockerLoginDockerHub)
+    // Docker Hub specific tasks
+    String dockerImageLoginDHTaskName = "dockerLogin${imageTaskNameSuffix}_DH"
+    String dockerImagePublishDHTaskName = "dockerPublish${imageTaskNameSuffix}_DH"
+
+    project.tasks.create(name: dockerImageLoginDHTaskName, type: Exec, group: 'Docker_DH', description: 'Docker login to hub.docker.com') {
+        dockerTasksCreated.add(dockerImageLoginDHTaskName)
+        outputs.upToDateWhen { false }
+
+        String dockerLoginUsername = System.env.DOCKER_INT_BLACKDUCK_USER ?: '<Not Set>'
+        doFirst {
+            String dockerLoginPassword = System.env.DOCKER_INT_BLACKDUCK_PASSWORD
+            if (dockerLoginPassword?.trim()) {
+                standardInput = new ByteArrayInputStream(dockerLoginPassword.getBytes())
+            }
+        }
+
+        commandLine 'docker', 'login', '--username', dockerLoginUsername, '--password-stdin'
+
+        onlyIf(envVarsNotExistMsg) {
+            System.env.DOCKER_INT_BLACKDUCK_USER != null && System.env.DOCKER_INT_BLACKDUCK_PASSWORD != null
+        }
+    }
+
+    project.tasks.create(name: dockerImagePublishDHTaskName, type: Exec, group: 'Docker_DH', description: "Docker push ${imageName} to hub.docker.com") {
+        dockerTasksCreated.add(dockerImagePublishDHTaskName)
+        outputs.upToDateWhen { false }
+        dependsOn project.tasks.findByName(dockerImageLoginDHTaskName)
+        commandLine 'docker', 'version'
```

Comment:
Is this just for testing at the moment?

Reply:
It is. I should have pointed that out. Until I am sure the logic is sound, I don't want to push any images to any repository